tests_pytest/common_tests/unit_tests/test_model_collector.py

```diff
@@ -68,13 +70,33 @@ def test_create_stats_collector_for_node_activation_disabled(self, fw_info_mock)
         """
         node = Mock()
         node.is_activation_quantization_enabled.return_value = False
+        node.is_quantization_preserving.return_value = False
+        node.is_fln_quantization.return_value = False
         node.type = DummyLayer
         # Even if prior_info exists, it should not be used.
         node.prior_info = Mock(min_output=None, max_output=None)
 
         collector = create_stats_collector_for_node(node, fw_info_mock)
         assert isinstance(collector, NoStatsCollector)
 
+    def test_create_stats_collector_for_node_fln_activation_enabled(self, fw_info_mock):
+        """
+        Verify that for a node with fln quantization enabled,
+        create_stats_collector_for_node returns a StatsCollector instance.
+        The test for FLN activation disabled is already covered by 
+            test_create_stats_collector_for_node_activation_disabled, so it is not checked here.
+        """
+        # Set up a fake node with activation quantization enabled and prior_info attributes.
+        node = Mock()
+        node.is_activation_quantization_enabled.return_value = False
+        node.is_quantization_preserving.return_value = False
+        node.is_fln_quantization.return_value = True
+        node.type = DummyLayer
+        node.prior_info = Mock(min_output=-8, max_output=9)
```

Comment:
this is a good use case that you are testing with prior_info. please also add a test scenario where the prior_info is None